src/v/storage/segment_appender.h

```diff
@@ -111,6 +111,21 @@ class segment_appender {
     ss::lw_shared_ptr<chunk> _head;
     ss::lw_shared_ptr<ss::semaphore> _prev_head_write;
 
+    struct flush_op {
+        explicit flush_op(size_t offset)
+          : offset(offset) {}
+        size_t offset;
+        ss::promise<> p;
+    };
+
+    std::vector<flush_op> _flush_ops;
```

Comment:
should we use seomthing like `boost::small_vector<flush_op, 6> _flush_ops;`

Reply:
without any evidence of workload profiles, I'm not sure.